extensions/amp-base-carousel/1.0/base-element.js

```diff
@@ -0,0 +1,120 @@
+ * limitations under the License.
+ */
+
+import {BaseCarousel} from './component';
+import {CSS as COMPONENT_CSS} from './component.jss';
+import {CarouselContextProp} from './carousel-props';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {dict} from '../../../src/utils/object';
+
+/** @extends {PreactBaseElement<BaseCarouselDef.CarouselApi>} */
+export class BaseElement extends PreactBaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?number} */
+    this.slide_ = null;
+  }
+
+  /** @override */
+  init() {
+    this.slide_ = parseInt(this.element.getAttribute('slide'), 10);
+
+    return dict({
+      'onSlideChange': (index) =>
+        this.triggerEvent(this.element, 'slideChange', dict({'index': index})),
+    });
+  }
+
+  /** @override */
+  mutationObserverCallback() {
+    const slide = parseInt(this.element.getAttribute('slide'), 10);
+    if (slide === this.slide_) {
+      return;
+    }
+    this.slide_ = slide;
+    if (!isNaN(slide)) {
+      this.api().goToSlide(slide);
+    }
+  }
+}
+
+/** @override */
+BaseElement['Component'] = BaseCarousel;
+
+/** @override */
+BaseElement['layoutSizeDefined'] = true;
+
+/** @override */
+BaseElement['children'] = {
+  'arrowPrev': {
+    name: 'arrowPrev',
+    selector: '[slot="prev-arrow"]',
+    single: true,
+  },
+  'arrowNext': {
+    name: 'arrowNext',
+    selector: '[slot="next-arrow"]',
+    single: true,
+  },
+  'children': {
+    name: 'children',
+    props: {
+      'thumbnailSrc': {attr: 'data-thumbnail-src'},
+    },
+    selector: '*', // This should be last as catch-all.
+    single: false,
+  },
+};
+
+/** @override */
+BaseElement['props'] = {
+  'advanceCount': {attr: 'advance-count', type: 'number', media: true},
+  'autoAdvance': {attr: 'auto-advance', type: 'boolean', media: true},
+  'autoAdvanceCount': {attr: 'auto-advance-count', type: 'number', media: true},
+  'autoAdvanceInterval': {
+    attr: 'auto-advance-interval',
+    type: 'number',
+    media: true,
+  },
+  'autoAdvanceLoops': {attr: 'auto-advance-loops', type: 'number', media: true},
+  'controls': {attr: 'controls', type: 'string', media: true},
+  'orientation': {
+    attr: 'orientation',
+    type: 'string',
+    media: true,
+    default: 'horizontal',
+  },
+  'loop': {attr: 'loop', type: 'boolean', media: true},
+  'mixedLength': {attr: 'mixed-length', type: 'boolean', media: true},
+  'outsetArrows': {attr: 'outset-arrows', type: 'boolean', media: true},
+  'snap': {attr: 'snap', type: 'boolean', media: true, default: true},
+  'snapBy': {attr: 'snap-by', type: 'number', media: true},
+  'snapAlign': {attr: 'snap-align', type: 'string', media: true},
+  'visibleCount': {attr: 'visible-count', type: 'number', media: true},
+};
+
+/** @override */
+BaseElement['staticProps'] = {
+  'slide': {attr: 'slide', type: 'number', default: '0'},
```

Comment:
```suggestion
  'defaultSlide': {attr: 'slide', type: 'number', default: '0'},
```

Reply:
Thank you! Looks like this was also caught by the tests, so that's a good sign

Reply:
Oh awesome!